File: modules/@angular/facade/src/browser.ts
/**
 * JS version of browser APIs. This library can only run in the browser.
 */
var win = typeof window !== 'undefined' && window || <any>{};

export {win as window};
export var document = win.document;
export var location = win.location;
export var gc = win['gc'] ? () => win['gc']() : (): any => null;
export var performance = win['performance'] ? win['performance'] : null;
export const Event = win['Event'];
export const MouseEvent = win['MouseEvent'];
export const KeyboardEvent = win['KeyboardEvent'];
export const EventTarget = win['EventTarget'];
export const History = win['History'];
export const Location = win['Location'];
export const EventListener = win['EventListener'];

File: modules/@angular/facade/src/browser.dart
/**
 * Dart version of browser APIs. This library depends on 'dart:html' and
 * therefore can only run in the browser.
 */
library angular2.src.facade.browser;

import 'dart:js' show context;
import 'dart:html' show Location, window;

export 'dart:html'
    show
        document,
        window,
        Element,
        Node,
        MouseEvent,
        KeyboardEvent,
        Event,
        EventTarget,
        History,
        Location,
        EventListener;

Location get location => window.location;

final _gc = context['gc'];

void gc() {
  if (_gc != null) {
    _gc.apply(const []);
  }
}
